client/landing/gutenboarding/hooks/use-on-site-creation.ts

```diff
@@ -75,41 +46,53 @@ export default function useOnSiteCreation(): void {
 	const flow = useOnboardingFlow();
 
 	const { resetOnboardStore, setIsRedirecting, setSelectedSite } = useDispatch( ONBOARD_STORE );
-	const flowCompleteTrackingParams = {
-		isNewSite: !! newSite,
-		isNewUser: !! newUser,
-		blogId: newSite?.blogid,
-		hasCartItems: false,
-	};
+	const flowCompleteTrackingParams = React.useMemo(
+		() => ( {
+			isNewSite: !! newSite,
+			isNewUser: !! newUser,
+			blogId: newSite?.blogid,
+			hasCartItems: false,
+		} ),
+		[ newSite, newUser ]
+	);
 
 	React.useEffect( () => {
 		// isRedirecting check this is needed to make sure we don't overwrite the first window.location.replace() call
 		if ( newSite && ! isRedirecting ) {
 			setIsRedirecting( true );
 
-			if ( selectedPlan && ! selectedPlan?.isFree ) {
-				const planProduct = {
-					product_id: planProductSource?.productId,
-					product_slug: planProductSource?.storeSlug,
+			if (
+				selectedPlan &&
+				! selectedPlan?.isFree &&
+				planProductSource &&
+				domain &&
+				domain.product_id &&
+				domain.product_slug
```

Comment:
@sirbrillig, I think this will prevent a paid plan from being added to cart if the user has skipped Domain step.